packages/tpetra/core/src/Tpetra_CrsMatrix_def.hpp

```diff
@@ -2434,6 +2436,7 @@ namespace Tpetra {
           // curOffset, the range is empty.
           const LO numIndInSeq = (endOffset - curOffset);
           if (numIndInSeq != 0) {
+            rowInfo = graph.getRowInfo(lclRow);  // KDD 5/19 Need fresh RowInfo in each loop iteration
```

Comment:
Not really.  `insertGlobalValuesImpl` actually updates its `rowInfo` argument (it's a read + write argument).  It should have the correct values on exit.  Calling `getRowInfo` can be a bit expensive, which is why the `insert*ValuesImpl` and other methods separate that out.